### PR TITLE
[backport] Fix `linalg.svd` for 0-sized matrices

### PR DESCRIPTION
**Pull request by takagi**
Backport of #3354

### CODE REVIEW

**Comment by takagi** [comment]
[automatic post] Jenkins, test this please.

File: tests/cupy_tests/linalg_tests/test_decomposition.py [hunk 1]
```diff
@@ -185,7 +188,7 @@ def check_singular(self, shape, xp, dtype):
         return result
 
     def check_rank2(self, array):
-        with self.assertRaises(numpy.linalg.LinAlgError):
+        with pytest.raises(numpy.linalg.LinAlgError):
```
Comment (toslunar) [inline tests/cupy_tests/linalg_tests/test_decomposition.py:191]: This line fixes the test to pytest style in the class.  Could you add `import pytest`? 
```diff
--- a/tests/cupy_tests/linalg_tests/test_decomposition.py
+++ b/tests/cupy_tests/linalg_tests/test_decomposition.py
@@ -1,6 +1,7 @@
 import unittest

 import numpy
+import pytest
 import six

 import cupy
```

Comment (takagi) [reply tests/cupy_tests/linalg_tests/test_decomposition.py:192]: Thanks, I've fixed!

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**